rs/tests/networking/canister_http_correctness_test.rs

```diff
@@ -610,30 +814,1006 @@ pub fn test_http_calls_to_ic_fails(env: TestEnv) {
+        },
+        HttpHeader {
+            name: "name2".to_string(),
+            value: "value2".to_string(),
+        },
+    ]);
+    let max_response_bytes = Some(666);
+
+    let request = CanisterHttpRequestArgs {
+        url,
+        headers,
+        method: HttpMethod::HEAD,
+        body,
+        transform: None,
+        max_response_bytes,
+    };
+
+    let response = block_on(submit_outcall(
+        &handlers,
+        RemoteHttpRequest {
+            request: request.clone(),
+            cycles: 500_000_000_000,
+        },
+    ))
+    .expect("Request succeeds.");
+
+    assert_matches!(&response, RemoteHttpResponse { status: 200, .. });
+    assert_distinct_headers(&response);
+}
+
+fn test_small_maximum_possible_response_size(env: TestEnv) {
+    let handlers = Handlers::new(&env);
+    let webserver_ipv6 = get_universal_vm_address(&env);
+
+    let url = format!(
+        "https://[{}]:20443/{}/{}",
+        webserver_ipv6, "ascii", "hello_world"
+    );
+
+    let header_size = 143;
+    let max_response_bytes = Some(header_size + "hello_world".len() as u64);
+
+    let response = block_on(submit_outcall(
+        &handlers,
+        RemoteHttpRequest {
+            request: CanisterHttpRequestArgs {
+                url,
+                headers: BoundedHttpHeaders::new(vec![]),
+                method: HttpMethod::GET,
+                body: None,
+                transform: None,
+                max_response_bytes,
+            },
+            cycles: 500_000_000_000,
+        },
+    ))
+    .expect("Request is successful.");
+
+    assert_matches!(&response, RemoteHttpResponse { status: 200, .. });
+    assert_http_response(&response);
+}
+
+fn test_small_maximum_possible_response_size_exceeded(env: TestEnv) {
+    let handlers = Handlers::new(&env);
+    let webserver_ipv6 = get_universal_vm_address(&env);
+
+    let url = format!(
+        "https://[{}]:20443/{}/{}",
+        webserver_ipv6, "ascii", "hello_world"
+    );
+
+    //   { Response headers
+    //       date: Jan 1 1970 00:00:00 GMT
+    //       content-type: application/octet-stream
+    //       content-length: 11
+    //       access-control-allow-origin: *
+    //       access-control-allow-credentials: true
+    //   }
+
+    let header_size = 143;
+    let max_response_bytes = Some(header_size + "hello_world".len() as u64 - 1);
+
+    let response = block_on(submit_outcall(
+        &handlers,
+        RemoteHttpRequest {
+            request: CanisterHttpRequestArgs {
+                url,
+                headers: BoundedHttpHeaders::new(vec![]),
+                method: HttpMethod::GET,
+                body: None,
+                transform: None,
+                max_response_bytes,
+            },
+            cycles: 500_000_000_000,
+        },
+    ));
+
+    assert_matches!(
+        response,
+        Err(RejectResponse {
+            reject_code: RejectCode::SysFatal,
+            ..
+        })
+    );
+}
```

Comment:
These two tests match the tests test_maximum_possible_value_of_max_response_bytes and test_maximum_possible_value_of_max_response_bytes_exceeded.